errai-ui/src/main/java/org/jboss/errai/ui/rebind/ioc/element/HasNamedAnnotation.java

```diff
@@ -60,22 +66,17 @@ public boolean equals(final Object obj) {
   }
 
   @Override
-  public boolean unsafeIsAnnotationPresent(final Class<? extends Annotation> annotation) {
-    return Named.class.equals(annotation);
-  }
-
-  @Override
-  public Annotation[] unsafeGetAnnotations() {
-    return new Annotation[] { named };
+  public Collection<MetaAnnotation> getAnnotations() {
+    return singleton(new JavaReflectionAnnotation(named));
   }
 
   @Override
   @SuppressWarnings("unchecked")
-  public <A extends Annotation> A unsafeGetAnnotation(final Class<A> annotation) {
-    if (unsafeIsAnnotationPresent(annotation)) {
-      return (A) named;
+  public Optional<MetaAnnotation> getAnnotation(final Class<? extends Annotation> annotation) {
```

Comment:
For this and a similar implementation I saw earlier, here is a trick you could try that is also null-safe for `annotation`:

    return Optional
              .ofNullable(annotation)
              .filter(a -> isAnnotationPresent(a))
              .map(a -> new JavaReflectionAnnotation(named));

Reply:
Actually after I applied the change you suggested I noticed that this method is not necessary anymore, because the default implementation in HasAnnotation produces the exact same result using the getAnnotations() implementation that this class provides.